TACC/Core-Portal
# task/WP-304: Fixed the placeholder text for search to be more accurately represented

## Overview
The search bar's placeholder text, specifically in Data Files, was not accurately represented based on navigation and possible search results.
1. Open Data Files.
2. Initially the search placeholder text might say "Search My Data(Frontera)".
3. After navigating into folders inside My Data(Frontera), the placeholder text remains the same regardless of file depth/navigation.
4. The user could mistake the search path for wrong one. For example:

- If I've navigated to the following file hierarchy: My Data(Frontera) > archive > jobs > 2023-11-20, the current search bar text placeholder would say "Search My Data(Frontera)".
- But if I try and search with the keyword "archive", I'll get no results.
- However, if I navigated back to My Data(Frontera) and tried to search the same keyword "archive", I'd get the folder result.

In other words, having the placeholder be the system name the entire time implies I can search the whole system for folders/files regardless of depth/hierarchy--which isn't the case.

To fix this, I've made the placeholder text more accurate.


## Related

* [WP-304](https://jira.tacc.utexas.edu/browse/WP-304)

## Changes

- Added code to check if the current path is homeDir. If user is in the homeDir, the search bar's text placeholder will use the appropriate system name. If the user is not in the homeDir, the current directory name will be displayed for the text placeholder.

## Testing

1. Go to cep.test (make sure your search index is properly set up)
2. Navigate to 'Data Files'
3. Select 'My Data(Frontera)' as your system
4. The initial text placeholder should say 'Search My Data(Frontera)'.
5. Try searching for a folder or file name you know is inside this system.
6. Navigate to a folder inside My Data(Frontera) and confirm that the placeholder text is the current directory of the folder you're in. 
7. Try searching for a folder/file that's higher in the hierarchy. It shouldn't return those results since you're deeper in the file tree.
8. See my examples below in the screenshots to compare.
9. When testing, make sure that the other Search bar's placeholder text and functionality still work. Site-wide search bar (cep.test/search) and Search bar for job history in "History" tab, for example.

## UI
![Screenshot 2023-11-20 at 3 13 13 PM](https://github.com/TACC/Core-Portal/assets/50084480/2eb2b7a3-a701-42ea-bcd8-296a10a11fa0)




![Screenshot 2023-11-20 at 3 14 10 PM](https://github.com/TACC/Core-Portal/assets/50084480/ba96b951-b5f6-40fd-b9a7-4115482ac8d5)




**Searching for 'jobs' keyword from My Data(Frontera) directory**
![Screenshot 2023-11-20 at 3 14 51 PM](https://github.com/TACC/Core-Portal/assets/50084480/5a22137d-fb82-4a9f-911b-2015ebe94621)




## Notes
